vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/util/ModelToVLogConverter.java

```diff
@@ -22,74 +24,135 @@
 
 import java.util.List;
 
+import org.apache.commons.lang3.Validate;
 import org.semanticweb.vlog4j.core.model.api.Atom;
 import org.semanticweb.vlog4j.core.model.api.Conjunction;
+import org.semanticweb.vlog4j.core.model.api.Predicate;
 import org.semanticweb.vlog4j.core.model.api.Rule;
 import org.semanticweb.vlog4j.core.model.api.Term;
-import org.semanticweb.vlog4j.core.model.api.TermType;
+import org.semanticweb.vlog4j.core.reasoner.RuleRewriteStrategy;
+
+
+/**
+ * Utility class with static methods for converting from
+ * {@code org.semanticweb.vlog4j.core.model} model objects to
+ * {@code karmaresearch.vlog} model objects
+ * 
+ * @author Irina Dragoste
+ *
+ */
+public final class ModelToVLogConverter {
 
-public class ModelToVLogConverter {
+	public static final String PREDICATE_ARITY_SUFFIX_SEPARATOR = "-";
+
+	private ModelToVLogConverter() {
+	}
 
 	public static karmaresearch.vlog.Term toVLogTerm(final Term term) {
-		// TODO treat null case: throw exception or return null?
-		// TODO visitor pattern
-		// FIXME perhaps declare enum class only in VLog project, and use
-		// karmaresearch.vlog.Term.EnumType dependency
-		// FIXME The names of variables and constants need to be appropriately modified
-		// so VLog recognises them as such.
-		// Notes: Upper case for variables; lower case for constants? enclose constants
-		// with <>?
-		final TermType type = term.getType();
-		final karmaresearch.vlog.Term.TermType vlogTermType = type == TermType.CONSTANT
-				? karmaresearch.vlog.Term.TermType.CONSTANT
-				: karmaresearch.vlog.Term.TermType.VARIABLE;
-		final karmaresearch.vlog.Term vLogTerm = new karmaresearch.vlog.Term(vlogTermType, term.getName());
-		return vLogTerm;
+		Validate.notNull(term);
+		final TermToVLogConverter termToVlogConverter = new TermToVLogConverter();
+		term.accept(termToVlogConverter);
+		return termToVlogConverter.getVlogTerm();

```

Comment:
the accept method should actually return this value directly! No need to store it internally or to fetch it would getVlogTerm()

Reply:
removed the getter and vLogTerm field